### PR TITLE
Documentation: upgrade to 2.3

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

**Comment by gyuho** [comment]
For before we launch etcd 2.3. Already reviewed by Josh.

@xiang90

File: Documentation/upgrade_2_3.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+To upgrade an existing etcd deployment to 2.3, you must be running 2.2. If you’re running a version of etcd before 2.2, you must upgrade to [2.2](https://github.com/coreos/etcd/releases/tag/v2.2.0) before upgrading to 2.3.
+
+Also, to ensure a smooth rolling upgrade, your running cluster must be healthy. You can check the health of the cluster by using the `etcdctl cluster-health` command.
+
+#### Preparation 
+
+Before upgrading etcd, always test the services relying on etcd in a staging environment before deploying the upgrade to the production environment. 
+
+Before begining,  [backup the etcd data directory](admin_guide.md#backing-up-the-datastore). Should something go wrong with the upgrade, it is possible to use this backup to[downgrade](#downgrade) back to your existing etcd version.
```
Comment (heyitsanthony) [inline Documentation/upgrade_2_3.md]: s/begining/beginning/

Comment (gyuho) [inline Documentation/upgrade_2_3.md]: thanks will fix.

File: Documentation/upgrade_2_3.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+#### Mixed Versions
+
+While upgrading, an etcd cluster supports mixed versions of etcd members, and operates with the protocol of the lowest common version. The cluster is only considered upgraded once all of its members are upgraded to version 2.3. Internally, etcd members negotiate with each other to determine the overall cluster version, which controls the reported version and the supported features.
+
+#### Limitations
+
+It might take up to 2 minutes for the newly upgraded member to catch up with the existing cluster when the total data size is larger than 50MB. Check the size of a recent  snapshot to estimate  the total data size. In other words, it is safest to wait for 2 minutes between upgrading each member..
```
Comment (heyitsanthony) [inline Documentation/upgrade_2_3.md]: s/.././

File: Documentation/upgrade_2_3.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+When each etcd process is stopped, expected errors will be logged by other cluster members. This is normal since a cluster member connection has been (temporarily) broken:
+
+```
+2016-03-11 09:50:49.860319 E | rafthttp: failed to read 8211f1d0f64f3269 on stream Message (unexpected EOF)
+2016-03-11 09:50:49.860335 I | rafthttp: the connection with 8211f1d0f64f3269 became inactive
+2016-03-11 09:50:51.023804 W | etcdserver: failed to reach the peerURL(http://127.0.0.1:12380) of member 8211f1d0f64f3269 (Get http://127.0.0.1:12380/version: dial tcp 127.0.0.1:12380: getsockopt: connection refused)
+2016-03-11 09:50:51.023821 W | etcdserver: cannot get the version of member 8211f1d0f64f3269 (Get http://127.0.0.1:12380/version: dial tcp 127.0.0.1:12380: getsockopt: connection refused)
+```
+
+It’s a good idea at this point to  [backup the etcd data directory](https://github.com/coreos/etcd/blob/7f7e2cc79d9c5c342a6eb1e48c386b0223cf934e/Documentation/admin_guide.md#backing-up-the-datastore) to provide a downgrade path should any problems occur:.
```
Comment (heyitsanthony) [inline Documentation/upgrade_2_3.md]: s/:./:

File: Documentation/upgrade_2_3.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+```
+$ etcdctl backup \
+      --data-dir /var/lib/etcd \
+      --backup-dir /tmp/etcd_backup
+```
+
+#### 3. Drop-in etcd v2.3 binary and start the new etcd process
+
+The new v2.3 etcd will publish its information to the cluster:.
```
Comment (heyitsanthony) [inline Documentation/upgrade_2_3.md]: s/:./:

File: Documentation/upgrade_2_3.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
+## Upgrade etcd from 2.2 to 2.3
+
+In the general case, upgrading from etcd 2.2 to 2.3 can be a zero-downtime, rolling upgrade:
+ - one by one, stop the etcd v2.2 processes and replace them with etcd v2.3 processes
+ - after you are running all v2.3 processes, new features in v2.3 are available to the cluster
```
Comment (heyitsanthony) [inline Documentation/upgrade_2_3.md]: eschew you?

File: Documentation/upgrade_2_3.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+Before [starting an upgrade](#upgrade-procedure), read through the rest of this guide to prepare.
+
+### Upgrade Checklists
+
+#### Upgrade Requirements
+
+To upgrade an existing etcd deployment to 2.3, you must be running 2.2. If you’re running a version of etcd before 2.2, you must upgrade to [2.2](https://github.com/coreos/etcd/releases/tag/v2.2.0) before upgrading to 2.3.
```
Comment (heyitsanthony) [inline Documentation/upgrade_2_3.md]: eschew you?

File: Documentation/upgrade_2_3.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+Also, to ensure a smooth rolling upgrade, your running cluster must be healthy. You can check the health of the cluster by using the `etcdctl cluster-health` command.
```
Comment (heyitsanthony) [inline Documentation/upgrade_2_3.md]: eschew you?

File: Documentation/upgrade_2_3.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+#### Preparation
+
+Before upgrading etcd, always test the services relying on etcd in a staging environment before deploying the upgrade to the production environment.
+
+Before beginning,  [backup the etcd data directory](admin_guide.md#backing-up-the-datastore). Should something go wrong with the upgrade, it is possible to use this backup to[downgrade](#downgrade) back to your existing etcd version.
```
Comment (heyitsanthony) [inline Documentation/upgrade_2_3.md]: eschew you?

**Comment by heyitsanthony** [comment]
lgtm aside from nits

File: Documentation/upgrade_2_3.md (hunk 8 above)
Comment (gyuho) [inline Documentation/upgrade_2_3.md]: Right. Will remove `you`s. thanks.

**Comment by xiang90** [comment]
LGTM

**Comment by gyuho** [comment]
will merge when we are about to release 2.3.

**Pull request merged**